### PR TITLE
Show correct variations count when generating a single variation

### LINKED ISSUE #37644: Variation count is wrong
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When creating a new product, after pressing `Generate variations` and there is only one variation to be created the count of variations is incorrect.

Screen Capture2 on 2023-04-10 at 16-19-21

### Expected behavior

The variation count should be correct 

### Actual behavior

The variation count now is showing a +1 variation.

### Steps to reproduce

1. Go to Products > Add New > Product data.
2. Select a `Variable product` and go to the `Attributes` tab.
3. Add a new attribute with only one value/term and save the attribute.
4. Go to the `Variations` tab and press `Generate variations`.
5. Verify that the variations count is incorrect
screenshot-one local-2023 04 10-16_25_41

### WordPress Environment

It's happening in `trunk`

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by mattsherman**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR corrects the variations count shown when using the "Generate variations" button and a single variation is generated.

#### Before

Screenshot 2023-04-20 at 08 47 03

#### After

Screenshot 2023-04-20 at 08 48 24

Closes #37644.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to `Products` > `Add New` > `Product data`.
2. Select a `Variable product` and go to the `Attributes` tab.
3. Add a new attribute with only one value/term and save the attribute.
4. Go to the `Variations` tab and click `Generate variations`.
5. Verify that the variations count is correct.
6. Also, use `Add manually` and `Remove` to add variations and verify the count remains correct.
7. Repeat this with an attribute with more than one value/term and verify that when multiple variations are generated, the count is correct.

### CODE REVIEW

File: plugins/woocommerce/client/legacy/js/admin/meta-boxes-product-variation.js [hunk 1]
```diff
@@ -1118,25 +1104,15 @@ jQuery( function ( $ ) {
...
-							window.alert(
-								woocommerce_admin_meta_boxes_variations.i18n_no_variations_added
-							);
-						}
+						const message =
+							count === 1
+								? woocommerce_admin_meta_boxes_variations.i18n_variation_added
+								: woocommerce_admin_meta_boxes_variations.i18n_variations_added;
+
+						window.alert( message.replace( '%qty%', count ) );
```
Comment (octaedro) [inline plugins/woocommerce/client/legacy/js/admin/meta-boxes-product-variation.js]: We don't need to replace the substring `%qty%` when we have only one variation.
Maybe we can do something like this:
```
const message = count === 1 ? i18n_variation_added : ( i18n_variations_added ).replace( '%qty%', count );
window.alert( message );
```

And also modify the `i18n_variation_added` text.

File: plugins/woocommerce/client/legacy/js/admin/meta-boxes-product-variation.js [hunk 2]
```diff
@@ -1118,25 +1104,15 @@ jQuery( function ( $ ) {
 					woocommerce_admin_meta_boxes_variations.ajax_url,
 					data,
 					function ( response ) {
-						var count = parseInt( response, 10 );
+						let count = parseInt( response, 10 );
+						count = isNaN( count ) ? 0 : count;
```
Comment (octaedro) [inline plugins/woocommerce/client/legacy/js/admin/meta-boxes-product-variation.js]: What do you think about doing something like this?
```
const count = parseInt( response, 10 ) || 0;
```

File: plugins/woocommerce/includes/admin/class-wc-admin-assets.php [hunk 3]
```diff
@@ -297,9 +297,8 @@ public function admin_scripts() {
 					'i18n_last_warning'                   => esc_js( __( 'Last warning, are you sure?', 'woocommerce' ) ),
 					'i18n_choose_image'                   => esc_js( __( 'Choose an image', 'woocommerce' ) ),
 					'i18n_set_image'                      => esc_js( __( 'Set variation image', 'woocommerce' ) ),
-					'i18n_variation_added'                => esc_js( __( 'variation added', 'woocommerce' ) ),
-					'i18n_variations_added'               => esc_js( __( 'variations added', 'woocommerce' ) ),
-					'i18n_no_variations_added'            => esc_js( __( 'No variations added', 'woocommerce' ) ),
+					'i18n_variation_added'                => esc_js( __( '%qty% variation added', 'woocommerce' ) ),
```
Comment (octaedro) [inline plugins/woocommerce/includes/admin/class-wc-admin-assets.php]: Please, see comment above.

**Review by octaedro — COMMENTED** [review]
Nice job @mattsherman. The changes look good, I only left a couple of comments.
Maybe I have a building problem, but I'm seeing the confirmation window like this:

Screenshot 2023-04-26 at 11 36 53

**Comment by mattsherman** [comment]
> Nice job @mattsherman. The changes look good, I only left a couple of comments. Maybe I have a building problem, but I'm seeing the confirmation window like this:
> 
> Screenshot 2023-04-26 at 11 36 53

This happens when the legacy JS isn't built. I just hit it when forgetting to `pnpm run build` after switching back to this branch.

File: plugins/woocommerce/client/legacy/js/admin/meta-boxes-product-variation.js (hunk 1 above)
Comment (mattsherman) [reply plugins/woocommerce/client/legacy/js/admin/meta-boxes-product-variation.js]: Fair point. 😄 Done!

File: plugins/woocommerce/includes/admin/class-wc-admin-assets.php (hunk 3 above)
Comment (mattsherman) [reply plugins/woocommerce/includes/admin/class-wc-admin-assets.php]: Done

File: plugins/woocommerce/client/legacy/js/admin/meta-boxes-product-variation.js (hunk 2 above)
Comment (mattsherman) [reply plugins/woocommerce/client/legacy/js/admin/meta-boxes-product-variation.js]: For sure... much better. Not sure what I was thinking?! 😄 Done.

**Comment by mattsherman** [comment]
@octaedro Ready for re-review!

**Review by octaedro — APPROVED** [review]
Great work @mattsherman! LGTM :shipit:

**Pull request merged**